src/pytest_split/plugin.py

```diff
@@ -1,14 +1,23 @@
 import json
 import os
 from collections import defaultdict, OrderedDict
+from typing import TYPE_CHECKING, Tuple
 
 from _pytest.config import create_terminal_writer
+from _pytest.main import Session
+
+if TYPE_CHECKING:
```

Comment:
If we introduce type hints, let's introduce type checking as well. For example, `mypy` pre-commit hook would be nice 🙂

Reply:
or is `flake8-type-checking` running mypy under the hood?

Reply:
No it doesn't - I agree, I'll add mypy!